pkg/virtctl/create/vm.go

```diff
@@ -0,0 +1,581 @@
+	cmd.Flags().StringArrayVar(&c.pvcVolumes, PvcVolumeFlag, c.pvcVolumes, fmt.Sprintf("Specify a PVCs to be used by the VM. Can be provided multiple times.\nSupported parameters: %s", supportedParams(pvcVolume{})))
+	cmd.Flags().StringArrayVar(&c.blankVolumes, BlankVolumeFlag, c.dataSourceVolumes, fmt.Sprintf("Specify a blank volume to be used by the VM. Can be provided multiple times.\nSupported parameters: %s", supportedParams(blankVolume{})))
+
+	cmd.Flags().StringVar(&c.cloudInitUserData, CloudInitUserDataFlag, c.cloudInitUserData, "Specify the base64 encoded cloud-init user data of the VM.")
+	cmd.Flags().StringVar(&c.cloudInitNetworkData, CloudInitNetworkDataFlag, c.cloudInitNetworkData, "Specify the base64 encoded cloud-init network data of the VM.")
+
+	cmd.Flags().SortFlags = false
+	cmd.SetUsageTemplate(templates.UsageTemplate())
+
+	return cmd
+}
+
+func defaultCreateVM() createVM {
+	return createVM{
+		name:                   "vm-" + rand.String(5),
+		terminationGracePeriod: 180,
+		runStrategy:            string(v1.RunStrategyAlways),
+	}
+}
+
+func (c *createVM) run(cmd *cobra.Command) error {
+	vm := newVM(c)
+	for _, flag := range flags {
+		if cmd.Flags().Changed(flag) {
+			if err := optFns[flag](c, vm); err != nil {
+				return err
+			}
+		}
+	}
+
+	out, err := yaml.Marshal(vm)
+	if err != nil {
+		return err
+	}
+
+	cmd.Print(string(out))
+	return nil
+}
+
+func (c *createVM) usage() string {
+	return `  # Create a manifest for a VirtualMachine with a random name:
+  {{ProgramName}} create vm
+    
+  # Create a manifest for a running VirtualMachine with a specified name
+  {{ProgramName}} create vm --name=my-vm --running=true
+
+  # Create a manifest for a VirtualMachine with a specified name and RunStrategy Always
+  {{ProgramName}} create vm --name=my-vm --run-strategy=Always
+  
+  # Create a manifest for a VirtualMachine with a specified VirtualMachineClusterInstancetype
+  {{ProgramName}} create vm --instancetype=my-instancetype
+
+  # Create a manifest for a VirtualMachine with a specified VirtualMachineInstancetype (namespaced)
+  {{ProgramName}} create vm --instancetype=virtualmachineinstancetype/my-instancetype
+  
+  # Create a manifest for a VirtualMachine with a specified VirtualMachineClusterPreference
+  {{ProgramName}} create vm --preference=my-preference
+
+  # Create a manifest for a VirtualMachine with a specified VirtualMachinePreference (namespaced)
+  {{ProgramName}} create vm --preference=virtualmachinepreference/my-preference
+
+  # Create a manifest for a VirtualMachine with a cloned DataSource in namespace and specified size
+  {{ProgramName}} create vm --volume-datasource=src:my-ns/my-ds,size:50Gi
+
+  # Create a manifest for a VirtualMachine with a cloned containerdisk
+  {{ProgramName}} create vm --volume-containerdisk=src:docker://my.registry/my-image:my-tag,size:50Gi
```

Comment:
I made a more detailed comment about this later down the file, but maybe we can call this ```--volume-import``` instead of ```volume-containerdisk``` in order to avoid people misinterpreting what the volume is since there's actually a volume source called ContainerDisk which this option doesn't relate to.